src/TestUtils/GcloudWrapper.php

```diff
@@ -117,11 +117,24 @@ public function setDir($dir)
      * @param int $retries optional Number of retries upon failure.
      * @return bool true if deployment suceeds, false upon failure.
      */
-    public function deploy(
-        $targets = 'app.yaml',
-        $promote = false,
-        $retries = self::DEFAULT_RETRIES
-    ) {
+    public function deploy($options = [])
+    {
+        // Handle old function signature
+        if (!is_array($options)) {
+            $options = array_filter([
+                'targets' => @func_get_arg(0),
+                'promote' => @func_get_arg(1),
+            ]) + array_filter([
+                'retries' => @func_get_arg(2),
+            ], 'is_numeric');
+        }
+        $options = array_merge([
+            'targets' => 'app.yaml',
+            'promote' => false,
+            'retries' => self::DEFAULT_RETRIES,
+            'beta' => false,
```

Comment:
Just take it a grain of salt. I'd use `use_gcloud_beta` or something instead of `beta` for clarity.

Or even more, change it to `gcloud_channel` and allow even something like `alpha`?

Reply:
good call!